framework/utils/randomUtils.py

```diff
@@ -34,45 +34,47 @@ class BoxMullerGenerator:
   """
     Iterator class for the Box-Muller transform
   """
-  def __init__(self):
+  def __init__(self,engine=None):
     """
       Constructor.
-      @ In, None
+      @ In, engine, instance, optional, random number generator
       @ Out, None
     """
-    self.queue = deque()
+    self.queue = defaultdict(deque)
+    self.engine = engine
```

Comment:
Does this member get used anywhere (i.e., does `BoxMullerGenerator.engine` get used?)

To me it looks like `engine` always gets passed in or is `None`, so maybe we don't need to pass it in to the constructor `__init__`.

Reply:
You are right, deleted this.